app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java

```diff
@@ -0,0 +1,149 @@
+        mFab.setOnClickListener(v ->
+                startActivity(new Intent(this.getContext(), FavouritesActivity.class))
+        );
+        return rootview;
+    }
+
+    /**
+     * This method checks for the favourites from preferences list
+     * and passes them to another method for dealing with the required view.
+     * @param sharedPreferences
+     */
+    private void checkFavs(SharedPreferences sharedPreferences) {
+        viewVisibility(sharedPreferences, pref_img_to_pdf,
+                getResources().getString(R.string.img_to_pdf));
+        viewVisibility(sharedPreferences, pref_text_to_pdf,
+                getResources().getString(R.string.text_to_pdf));
+        viewVisibility(sharedPreferences, pref_qr_barcode,
+                getResources().getString(R.string.qr_barcode_pdf));
+        viewVisibility(sharedPreferences, pref_view_files,
+                getResources().getString(R.string.viewFiles));
+        viewVisibility(sharedPreferences, pref_history,
+                getResources().getString(R.string.history));
+        viewVisibility(sharedPreferences, pref_add_password,
+                getResources().getString(R.string.add_password));
+        viewVisibility(sharedPreferences, pref_rem_pass,
+                getResources().getString(R.string.remove_password));
+        viewVisibility(sharedPreferences, pref_rot_pages,
+                getResources().getString(R.string.rotate_pages));
+        viewVisibility(sharedPreferences, pref_add_watermark,
+                getResources().getString(R.string.add_watermark));
+        viewVisibility(sharedPreferences, pref_add_images,
+                getResources().getString(R.string.add_images));
+        viewVisibility(sharedPreferences, pref_merge_pdf,
+                getResources().getString(R.string.merge_pdf));
+        viewVisibility(sharedPreferences, pref_invert_pdf,
+                getResources().getString(R.string.invert_pdf));
+        viewVisibility(sharedPreferences, pref_compress,
+                getResources().getString(R.string.compress_pdf));
+        viewVisibility(sharedPreferences, pref_rem_dup_pages,
+                getResources().getString(R.string.remove_duplicate_pages));
+        viewVisibility(sharedPreferences, pref_remove_pages,
+                getResources().getString(R.string.remove_pages));
+        viewVisibility(sharedPreferences, pref_reorder_pages,
+                getResources().getString(R.string.reorder_pages));
+        viewVisibility(sharedPreferences, pref_extract_img,
+                getResources().getString(R.string.extract_images));
+        viewVisibility(sharedPreferences, pref_pdf_to_img,
+                getResources().getString(R.string.pdf_to_images));
+    }
+
+    /**
+     * This method toggles the visibility of the passed view.
+     * @param sharedPreferences
+     * @param view
+     * @param id
+     */
+    private void viewVisibility(SharedPreferences sharedPreferences, View view, String id) {
```

Comment:
no need to receive SharedPreferences parameter

Reply:
Actually we need it to get the boolean value for a given id

Reply:
You will directly be able to use `mSharedPreferences` variable. There wont be any need to get this as a parameter

Reply:
Yeah I got it now. I had misunderstood you.